git-subhistory.sh

```diff
@@ -351,7 +356,7 @@ cd ./$(git rev-parse --show-cdup) || exit $?
 GIT_DIR="$(git rev-parse --git-dir)"
 
 # a temporary directory for e.g. filter-branch backups
-mkdir "$GIT_DIR/subhistory-tmp/" || exit $?
+mkdir -p "$GIT_DIR/subhistory-tmp/" || exit $?
```

Comment:
The `-p` option does two things: create parent directories and does not error if the target directory already exists.  I wasn't thinking about the `.git` directory not existing, but I did run into situations in testing where the script failed to start because `.git/subhistory-tmp` already existed - I can't remember how it happened, and it may have been me quitting it by hitting "Ctrl-C".  It probably makes more sense to handle the directory already existing more explicitly - either print an error message explaining the problem and how to fix it, silently removing the old contents, or just silently moving on (like it does now).